src/pngimage.cpp

```diff
@@ -533,16 +552,17 @@ namespace Exiv2 {
         // Write PNG Signature.
         if (outIo.write(pngSignature, 8) != 8) throw Error(kerImageWriteFailed);
 
-        DataBuf cheaderBuf(8);       // Chunk header : 4 bytes (data size) + 4 bytes (chunk type).
+        DataBuf cheaderBuf(9);   // Chunk header : 4 bytes (data size) + 4 bytes (chunk type) + nul
+        cheaderBuf.pData_[8] = 0;
 
         while(!io_->eof())
         {
             // Read chunk header.
 
             std::memset(cheaderBuf.pData_, 0x00, cheaderBuf.size_);
-            long bufRead = io_->read(cheaderBuf.pData_, cheaderBuf.size_);
+            long bufRead = io_->read(cheaderBuf.pData_, 8);
```

Comment:
In case we additional byte is not needed at the end, I would revert this change and use `cheaderBuf.size_` instead of `8` (which could be considered as a magic value)